src/quickunlock/Polkit.cpp

```diff
@@ -128,12 +148,16 @@ bool Polkit::getKey(const QString& dbPath, QByteArray& key)
         }
 
         QByteArray keychainBytes(static_cast<const char*>(keychainBuffer), keychainDataSize);
+
+        free(keychainBuffer);
```

Comment:
Since you're scrubbing `keychainBytes`, you should also scrub `keychainBuffer` before freeing it.